src/mlmc/quantity_concept.py

```diff
@@ -121,120 +325,451 @@ def mult_op(x, y):
-    but can be of any type. Meanigful are only onther types then TimeSeries.
-    """
-    def __init__(self, time_frames : List[Tuple[float, QBase]]):
-        super().__init__(time_frames)
-        self.frame0 = time_frames[0][1]
-        assert( all( (q._size == self.frame0._size for t,q in time_frames) ) )
-        assert( all( (q.__class__ == self.frame0.__class__ for t,q in time_frames) ))
+    def __iter__(self):
+        raise Exception("This class is not iterable")
 
+    def __copy__(self):
+        return Quantity(quantity_type=self.qtype, input_quantities=custom_copy(self._input_quantities),
+                        operation=self._operation)
 
-        self.time_frames = time_frames
-        # list of pairs (time, value)
-        self.block_starts = []
-        # starts of blocks of the time values on the single sample row
-        self.operation = lambda lch : np.concatenate(lch, axis=0)
+    def time_interpolation(self, value):
+        """
+        Interpolation in time
+        :param value: point where to interpolate
+        :return: Quantity
+        """
+        def interp(y):
+            split_indeces = np.arange(1, len(self.qtype._times)) * self.qtype._qtype.size()
+            y = np.split(y, split_indeces, axis=-3)
+            f = interpolate.interp1d(self.qtype._times, y, axis=0)
+            return f(value)
+
+        return Quantity(quantity_type=self.qtype._qtype, input_quantities=custom_copy([self]), operation=interp)
 
-    def max(self):
+    def level_ids(self):
         """
-        Create quantity for the maximum over the time series.
-        :return:
+        List of level ids, all input quantities must be from the same storage,
+        so getting level IDs from one of them should be completely fine
+        :return: List[int]
         """
-        def max_op(*input_chunks):
-            table = np.stack(input_chunks)
-            return np.max(table, axis=0)
+        return self._input_quantities[0].level_ids()
+
 
-        return self.frame0._copy(*[q for t, q in self.time_frames], operation = max_op)
+class QuantityMean:
 
-    def interpolate(self):
+    def __init__(self, quantity_type, mean, var):
         """
-        Create quantity for the maximum over the time series.
+        QuantityMean represents result of estimate_mean method
+        :param quantity_type: QType
+        :param mean: np.ndarray
+        :param var: np.ndarray
+        """
+        self.qtype = quantity_type
+        self._mean = mean
+        self._var = var
+
+    def __call__(self):
+        """
+        Return mean
         :return:
         """
+        return self.mean()
+
+    def var(self):
+        return self._var
 
-        def max_op(*input_chunks):
-            table = np.stack(input_chunks)
-            return np.max(table, axis=0)
+    def mean(self):
+        return self._mean
 
-        return self.frame0._copy(*[q for t, q in self.time_frames], operation=max_op)
+    def __getitem__(self, key):
```

Comment:
Should be reused from quantity or rather from QType descendants. See the note above.